src/error.rs

```diff
@@ -82,6 +79,8 @@ impl Display for OxidebpfError {
     }
 }
 
+impl std::error::Error for OxidebpfError {}
```

Comment:
this is standard in rust errors. It let's people that just want a generic error do: `Box<dyn std::error::Error>`